boot/bootutil/include/bootutil/image.h

```diff
@@ -64,28 +68,28 @@ struct image_version {
     uint32_t iv_build_num;
 };
 
-#define IMAGE_SIZE(hdr)                                                 \
-    ((hdr)->ih_tlv_size + (hdr)->ih_hdr_size + (hdr)->ih_img_size)
-
 /** Image header.  All fields are in little endian byte order. */
 struct image_header {
     uint32_t ih_magic;
-    uint16_t ih_tlv_size; /* Combined size of trailing TLVs (bytes). */
-    uint8_t  ih_key_id;   /* Which key image is signed with (0xff=unsigned). */
-    uint8_t  _pad1;
+    uint16_t _pad1;
     uint16_t ih_hdr_size; /* Size of image header (bytes). */
-    uint16_t _pad2;
-    uint32_t ih_img_size; /* Does not include header. */
+    uint32_t ih_img_size; /* Does not include header or TLVs. */
     uint32_t ih_flags;    /* IMAGE_F_[...]. */
     struct image_version ih_ver;
-    uint32_t _pad3;
+    uint32_t _pad2[2];
+};
+
+/** Image TLV info. All fields in little endian. */
+struct image_tlv_info {
+    uint16_t it_magic;
+    uint16_t it_tlv_tot;  /* size of TLV area (including tlv_info header) */
 };
 
 /** Image trailer TLV format. All fields in little endian. */
 struct image_tlv {
-    uint8_t  it_type;   /* IMAGE_TLV_[...]. */
+    uint8_t  it_type;     /* IMAGE_TLV_[...]. */
```

Comment:
Since we're already changing things, why don't we make the type field 16-bits instead of 8. It seems kind of silly to have padding there, and then have a very small tag field that could easily overflow.

Reply:
That we could do.